lib/internal/test_runner/test.js

```diff
@@ -776,7 +783,7 @@ class Test extends AsyncResource {
   }
 
   reportStarted() {
-    if (this.#reportedSubtest || this.parent === null) {
+    if (this.#reportedSubtest || this.parent === null || this.root.bailedOut) {
```

Comment:
we should prevent execution of tests, not just reporting them

Reply:
Probably this is no longer necessary